catalyst/core/runner.py

```diff
@@ -318,7 +318,11 @@ def _run_epoch(self, stage: str, epoch: int):
             self.state.loader_name = loader_name
             self.state.loader_len = len(loader)
             self.state.need_backward = loader_name.startswith("train")
-            self.model.train(self.state.need_backward)
+            if isinstance(self.model, dict):
```

Comment:
what about `maybe_recursive_call`?

Reply:
Thank you, tested it right now - works like a charm.

Reply:
do we really need such if in the core runner?
from Catalyst typing Model is instance of nn.Module... and maybe nn.ModuleDict, but not dict itself